### DISCUSSION TITLE
3D collision?

### DISCUSSION THREAD

**Discussion by ostix360** [dyn4j/dyn4j#203] [Q&A]
Is it hard to change this lib on a 3D collision detection?
I juste need to add the z coordinate in all the code? 
It's also a suggestion 
Thanks

**Comment by wnbittle** [comment]
If you only need collision detection, then it's easier, but it won't be as easy as adding a z coordinate.  Rotations will need to be stored differently (quat), positions need a new z coordinate as mentioned, the broadphase and narrowphase detection algorithms need to be extended to 3D, all the shapes would need to be changed, there are a lot of 2D specific optimizations that would need to be revisited, just to name a few things.

You would need to revise the following packages in particular:

- `org.dyn4j.geometry`
- `org.dyn4j.collision`
- `org.dyn4j.collision.broadphase`
- `org.dyn4j.collision.narrowphase`
- `org.dyn4j.collision.manifold`
